packages/astro-ghostcms/src/astro-ghostcms.ts

```diff
@@ -0,0 +1,315 @@
+import fse from "fs-extra";
+import { createResolver, defineIntegration } from "astro-integration-kit";
+import { corePlugins } from "astro-integration-kit/plugins";
+import { AstroError } from "astro/errors";
+import type { AstroIntegration } from "astro";
+import c from "picocolors";
+import { loadEnv } from "vite";
+import sitemap from "@astrojs/sitemap";
+import robotsTxt from "astro-robots-txt";
+
+// Internal Imports
+import { GhostUserConfigSchema } from "./schemas/userconfig";
+import ghostRSS from "./integrations/rssfeed";
+import ghostOGImages from "./integrations/satoriog";
+import ghostThemeProvider from "./integrations/themeprovider";
+import latestVersion from "./utils/latestVersion";
+
+// Load environment variables
+const ENV = loadEnv("all", process.cwd(), "CONTENT_API");
+
+/** Astro-GhostCMS Integration
+ * @description This integration allows you to use GhostCMS as a headless CMS for your Astro project
+ * @see https://astro-ghostcms.xyz for the most up-to-date documentation!
+ */
+export default defineIntegration({
+	name: "@matthiesenxyz/astro-ghostcms",
+	optionsSchema: GhostUserConfigSchema,
+	plugins: [...corePlugins],
+	setup({ options, name }) {
+		const { resolve } = createResolver(import.meta.url);
+
+		return {
+			"astro:config:setup": ({
+				watchIntegration,
+				hasIntegration,
+				addIntegration,
+				addVirtualImports,
+				injectRoute,
+				logger,
+			}) => {
+				// Set up verbose logging
+				const verbose = options.verbose;
+
+				// Configure Loggers
+				const GhostLogger = logger.fork(c.bold(c.blue("👻 Astro-GhostCMS")));
+
+				// Configure ENV Logger
+				const GhostENVLogger = logger.fork(
+					`${c.bold(c.blue("👻 Astro-GhostCMS"))}${c.gray("/")}${c.blue(
```

Comment:
This logging function could also be extracted

```ts
const formattedMessage = message => `${c.bold(c.blue("👻 Astro-GhostCMS"))}${c.gray("/")}${c.blue(message)}`
```

Reply:
Updated...

```ts

				// Configure Loggers
				const GhostLogger = logger.fork(c.bold(c.blue("👻 Astro-GhostCMS")));

				const loggerTagged = (message: string) => {
					return logger.fork(`${c.bold(c.blue("👻 Astro-GhostCMS"))}${c.gray("/")}${c.blue(message)}`)
				}

				// Configure ENV Logger
				const GhostENVLogger = loggerTagged("ENV Check");

				// Configure Integration Loggers & verbose logging
				const GhostIntegrationLogger = loggerTagged("Integrations");

				// Configure Route Logger & verbose logging
				const GhostRouteLogger = loggerTagged("Router");
```